rust/src/array_ops.rs

```diff
@@ -155,6 +155,34 @@ where
     n
 }
 
+/// Returns the sum of values in the array.
+///
+/// Returns `None` if the array is empty or only contains null values.
+pub fn sum<T>(array: &PrimitiveArray<T>) -> Option<T::Native>
+where
+    T: ArrowNumericType,
+    T::Native: Add<Output = T::Native>,
+{
+    let mut n: T::Native = T::default_value();
+    // iteratively track whether all values are null (or array is empty)
+    let mut all_nulls = true;
+    let data = array.data();
+    for i in 0..data.len() {
+        if data.is_null(i) {
+            continue;
+        }
+        if all_nulls {
+            all_nulls = false;
+        }
+        let m = array.value(i);
+        n = n + m;
+    }
+    if all_nulls {
```

Comment:
I think this would be clearer as a single expression without using a `return` statement:

```
if all_nulls {
  None
} else {
  Some(n)
}

Reply:
Fixed, don't know what I was trying to do there